sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -853,6 +855,82 @@ def peek_messages(
         with receive_trace_context_manager(self, span_name=SPAN_NAME_PEEK, links=links, start_time=start_time):
             return messages
 
+    def delete_messages(
+        self,
+        *,
+        max_message_count: Optional[int] = None,
+        before_enqueued_time_utc: Optional[datetime.datetime] = None,
+    ) -> int:
+        """
+        This operation deletes messages in the queue that are older than the specified enqueued time.
+
+        :keyword int or None max_message_count: The maximum number of messages to delete.
+        :keyword datetime.datetime or None before_enqueued_time_utc: The UTC datetime value before which all messages
+         should be deleted. The default value is None, meaning all messages in the queue will be considered.
+        :rtype: int
+
+        """
+        self._check_live()
+        self._open()
+
+        message_count = max_message_count if max_message_count else 4000
+
+        message = {
+            MGMT_REQUEST_ENQUEUED_TIME_UTC: before_enqueued_time_utc if before_enqueued_time_utc
+                else datetime.datetime.now(datetime.timezone.utc),
+            MGMT_REQUEST_MAX_MESSAGE_COUNT: message_count,
+        }
+
+        self._populate_message_properties(message)
+        handler = functools.partial(mgmt_handlers.batch_delete_op, receiver=self, amqp_transport=self._amqp_transport)
+        start_time = time.time_ns()
+        deleted = self._mgmt_request_response_with_retry(
+            REQUEST_RESPONSE_DELETE_BATCH_OPERATION, message, handler
+        )
+
+        links = get_receive_links(deleted)
+        with receive_trace_context_manager(self, span_name=SPAN_NAME_PEEK, links=links, start_time=start_time):
+            return deleted
+
+    def purge_messages(
+        self,
+        *,
+        before_enqueued_time_utc: Optional[datetime.datetime] = None,
+    ) -> int:
+        """
+        This operation purges all messages in the queue that are older than the specified enqueued time.
```

Comment:
should this say something like, `if possible`. Since this feels like a guarantee that all messages will be deleted, but there might be scenarios where the service will return 0 but there are still locked messages in the q?